### PR TITLE
run tests on browserstack, tidy up some of the tests/make more robust

### PR DESCRIPTION
**Pull request by earnubs**
Run acceptance tests on browserstack, with junit reporter.

TODO

- [x] test suite working in ff and chrome on win10
- [x] junit reporter
- [x] multi reporters https://github.com/mochajs/mocha/pull/2184 (blocked?) or docs to specify reporter
- [x] jenkaas job to run tests hourly/daily
 
- [x] mocha parallel
- [x] define capabilities for ff, chrome & safari on win10 & macOS
Edge driver is blocked on https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/5238133/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,10 +45,13 @@ See [Environment Variables](docs/environment-variables.md).
...
-
-Install chromedriver https://sites.google.com/a/chromium.org/chromedriver/downloads
-With a running dev server at http://localhost:3000:
+Acceptance tests run on browserstack, to run locally you will need to set some
+environment variables (secrets available from project owner):
 
 ```
-TEST_USER_EMAIL=email  TEST_USER_PASSWORD=password npm run test:acceptance
+TEST_USER_EMAIL=email@example.com  TEST_USER_PASSWORD=password \
+BROWSERSTACK_USER=foo BROWSERSTACK_KEY=bar \
```
Comment (vilagithub) [inline README.md]: That's _USERNAME ;)

File: test/acceptance/driver.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+const capabilities = {
+  'project': 'my.staging.ubuntu.com',
+  'browserstack.user': BS_USER,
+  'browserstack.key': BS_KEY,
+  'browserName': 'Firefox',
+  'browser_version': '50',
+  'os': 'Windows',
+  'os_version': '10',
+  //'browserstack.debug': true
```
Comment (vilagithub) [inline test/acceptance/driver.js]: <testsuite name="authenticated session" timestamp="2016-12-14T21:28:10" tests="14" failures="0" time="54.465">

( with s/_USER/_USERNAME)

**Comment by earnubs** [comment]
[E2E Tests](https://trello.com/c/a9Fo4goC/79-e2e-tests)

**Comment by earnubs** [comment]
mocha parallel and webdriver mocha would need a bit more work than is in scope for this pr

**Comment by earnubs** [comment]
safari is blocked on an option select bug, which i'll address in a later branch, edge is blocked on a driver bug, noted in source

**Pull request merged**